### PR TITLE
cmd/img_mgmt/port/zephyr: fix to img_mgmt_impl_erase_image_data()

### PR DESCRIPTION
**Pull request by nvlsianpu**
It was possible that the function tried to eras non erase-block-size
aligned number of bytes witch might have cause failure.
Additionally the image trailer area might have been not erased at all.

Reworked implementation so:
 - the erase request is corrected to erase proper amount of memory
 - erase of image trailer is ensured.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
@de-nordic Wll send to the upstera after review.

**Comment by de-nordic** [comment]
Fix the commit message typos: eras -> erase, witch -> which . Thanks.

**Comment by de-nordic** [comment]
Here is review https://github.com/apache/mynewt-mcumgr/pull/80, please update your west.yml in Zephyr. Thanks.

**Comment by nvlsianpu** [comment]
DNM as we want to merge this upstream first.

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c [hunk 1]
```diff
@@ -318,18 +318,67 @@ img_mgmt_impl_erase_image_data(unsigned int off, unsigned int num_bytes)
...
 
+    if (off != 0) {
+        rc = MGMT_ERR_EINVAL;
+        goto end;
+    }
+
     rc = flash_area_open(FLASH_AREA_ID(image_1), &fa);
     if (rc != 0) {
-        return MGMT_ERR_EUNKNOWN;
+        LOG_ERR("Can't to bind to the flash area (err %d)", rc);
```
Comment (de-nordic) [inline cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:328]: unneeded to before bind.

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c [hunk 2]
```diff
@@ -318,18 +318,67 @@ img_mgmt_impl_erase_image_data(unsigned int off, unsigned int num_bytes)
...
+
+        rc = flash_area_erase(fa, off, erase_size);
+        if (rc != 0) {
+            LOG_ERR("image slot trailer erase of 0x%x bytes failed (err %d)",
+                    erase_size, rc);
+            rc = MGMT_ERR_EUNKNOWN;
+            goto end_fa;
+        }
+
+        LOG_INF("Erased 0x%x bytes of image slot trailer ", erase_size);
```
Comment (de-nordic) [inline cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:373]: space at the end of message.

**Comment by nvlsianpu** [comment]
close as is replaced by #23

**Pull request closed without merging**